addons/io_hubs_addon/components/definitions/media_frame.py

```diff
@@ -0,0 +1,211 @@
+        default=True
+    )
+
+    @classmethod
+    def update_gizmo(cls, ob, gizmo):
+        loc, rot, _ = ob.matrix_world.decompose()
+        scale = gizmo.target_get_value("bounds")
+        mat_out = Matrix.Translation(
+            loc) @ rot.normalized().to_matrix().to_4x4() @ Matrix.Diagonal(scale).to_4x4()
+        gizmo.matrix_basis = mat_out
+        gizmo.hide = not ob.visible_get()
+
+    @classmethod
+    def create_gizmo(cls, ob, gizmo_group):
+        widget = gizmo_group.gizmos.new(MediaFrameGizmo.bl_idname)
+        widget.object = ob
+        setattr(widget, "hubs_gizmo_shape", box.SHAPE)
+        widget.setup()
+        widget.use_draw_scale = False
+        widget.use_draw_modal = True
+        widget.color = (0.0, 0.0, 0.8)
+        widget.alpha = 1.0
+        widget.hide = not ob.visible_get()
+        widget.scale_basis = 1.0
+        widget.hide_select = False
+        widget.color_highlight = (0.0, 0.0, 0.8)
+        widget.alpha_highlight = 0.5
+
+        widget.target_set_prop(
+            "bounds", ob.hubs_component_media_frame, "bounds")
+
+        return widget
+
+    @classmethod
+    def migrate(cls, version):
+        migrate_networked(cls.get_name())
+
+        if version[1] < 1:
+            for ob in bpy.data.objects:
+                if cls.get_name() in ob.hubs_component_list.items:
+                    bounds = ob.hubs_component_media_frame.bounds.copy()
+                    bounds = Vector((bounds.x, bounds.z, bounds.y))
+                    ob.hubs_component_media_frame.bounds = bounds
+
+    @staticmethod
+    def register():
+        bpy.utils.register_class(MediaFrameGizmo)
+
+    @staticmethod
+    def unregister():
+        bpy.utils.unregister_class(MediaFrameGizmo)
+
+    def gather(self, export_settings, object):
+        bounds = {
+            'x': self.bounds.x,
+            'y': self.bounds.y,
+            'z': self.bounds.z
+        }
+        if export_settings['gltf_yup']:
+            bounds['y'] = self.bounds.z
+            bounds['z'] = self.bounds.y
+
+        return {
+            'bounds': bounds,
+            'mediaType': self.mediaType,
+            'snapToCenter': self.snapToCenter
+        }
+
+    def draw(self, context, layout):
+        super().draw(context, layout)
+
+        parents = [context.object]
+        while parents:
+            parent = parents.pop()
+            if parent.scale != V_S1:
+                col = layout.column()
+                col.alert = True
+                col.label(
+                    text="The media-frame object, and it's parents, scale needs to be [1,1,1]", icon='ERROR')
+
+                break
+
+            if parent.parent:
+                parents.insert(0, parent.parent)
+
+            if parent.parent_bone:
+                parents.insert(0, parent.parent_bone)
```

Comment:
Sorry, I made a mistake with this.
This should fix it:
```suggestion
            if hasattr(parent, 'parent_bone') and parent.parent_bone:
                parents.insert(0, parent.parent.pose.bones[parent.parent_bone])
```

Reply:
Updated